src/components/common/Tooltip/Tooltip.tsx

```diff
@@ -1,29 +1,34 @@
+import { useState } from 'react';
 import styles from './tooltip.module.scss';
 
 type TooltipProps = {
-    isVisible: boolean;
-    textToShow: string;
-    tooltipPosition: {
-        top?: string;
-        right?: string;
-        bottom?: string;
-        left?: string;
-    };
+    children: React.ReactNode;
+    content: React.ReactNode;
+    tooltipPosition?: React.CSSProperties;
 };
 
-export default function Tooltip({
-    isVisible,
-    textToShow,
-    tooltipPosition,
-}: TooltipProps) {
+const Tooltip = ({ children, content, tooltipPosition }: TooltipProps) => {
+    const [currClass, setCurrClass] = useState('fade-out');
```

Comment:
We can set useState type here to be `fade-out` or `fade-in` only

Reply:
Accpted